scene/resources/gradient.cpp

```diff
@@ -225,3 +228,15 @@ Color Gradient::get_color(int pos) {
 int Gradient::get_point_count() const {
 	return points.size();
 }
+
+Color Gradient::interpolate(const Color &p_from, const Color &p_to, float p_weight, ColorSpace p_interpolation_color_space, InterpolationMode p_interpolation_mode) {
+	Ref<Gradient> gradient;
+	gradient.instantiate();
+	gradient->set_color(0, p_from);
+	gradient->set_offset(0, 0.0f);
+	gradient->set_color(1, p_to);
+	gradient->set_offset(1, 1.0f);
+	gradient->set_interpolation_color_space(p_interpolation_color_space);
+	gradient->set_interpolation_mode(p_interpolation_mode);
+	return gradient->get_color_at_offset(p_weight);
+}
```

Comment:
There's bound to be quite a lot of overhead with this approach compared to `Color::lerp()`, especially due to the object instantiation. I wonder if we can reduce it somehow, considering this method may be called hundreds or even thousands of times per frame (e.g. if you have lots of actors and each of them is calling it a few times every frame).

Reply:
Is Gradient::interpolate static? One way is to leave the lifecycle of the object to the caller so they can reuse the object.

Reply:
Yes, it's a static method.

Reply:
tl;dr So to avoid instantiating too many objects, let the caller keep the object reference and avoid churning. Like remove the static flag.

Reply:
Could we keep alive a static gradient object instead?

```cpp
if (static_gradient.is_null()) {
    static_gradient.instantiate();
}
<reuse the existing PR code>
```

Reply:
I setup a `Gradient::initialize()` and `Gradient::deinitialize()` in order to reuse the same gradient over-and-over.

Reply:
I finally reverted back, removed `initialize()` and `uninitialize()`. Opted for the simple `is_null()` check with the `thread_local` qualifier on `_interpolation_gradient`.